unsafe/src/main/java/org/apache/spark/unsafe/types/Interval.java

```diff
@@ -43,12 +43,13 @@
    * Finally is the unit name, ends with an optional "s".
    */
   private static String unitRegex(String unit) {
-    return "(?:\\s+(-?\\d+)\\s+" + unit + "s?)?";
+    return "(?:\\s+(-?\\d+)\\s+" + unit + "s?\\s*)?";
```

Comment:
This class is the internal representation of interval type, written in Java for better performance, just like what we did for `UTF8String`,  that's why it's in 'unsafe'.
Currently our date and timestamp internal representation is int and long, which is more efficient than Joda time I think.